test/hotspot/jtreg/runtime/ConstantPool/BadMethodHandles.java

```diff
@@ -23,188 +23,163 @@
+                                            .ldc("hello from m")
+                                            .invokevirtual(CD_PrintStream, "println", MethodTypeDesc.of(CD_void, CD_String))
+                                            .return_())
+                            .withMethodBody("staticM", MTD_void, ACC_PUBLIC | ACC_STATIC,
+                                    cob -> cob
+                                            .getstatic(CD_System, "out", CD_LPrintStream)
+                                            .ldc("hello from staticM")
+                                            .invokevirtual(CD_PrintStream, "println", MethodTypeDesc.of(CD_void, CD_String))
+                                            .return_())
+                            .withMethodBody("runm", MTD_void, ACC_PUBLIC | ACC_STATIC,
+                                    cob -> cob
+                                            .ldc(handle_1)
+                                            .invokevirtual(CD_MethodHandle, "invoke", MethodTypeDesc.of(CD_void))
+                                            .return_())
+                            .withMethodBody("runStaticM", MTD_void, ACC_PUBLIC | ACC_STATIC,
+                                    cob -> cob
+                                            .ldc(handle_2)
+                                            .invokevirtual(CD_MethodHandle, "invoke", MethodTypeDesc.of(CD_void))
+                                            .return_())
 
-        cw.visitEnd();
-        return cw.toByteArray();
+        );
+
+        return bytes;
     }
 
     static byte[] dumpIBad() {
-        ClassWriter cw = new ClassWriter(0);
-        cw.visit(52, ACC_PUBLIC | ACC_ABSTRACT | ACC_INTERFACE, "IBad", null, "java/lang/Object", null);
-        {
-            MethodVisitor mv = cw.visitMethod(ACC_PUBLIC, "m", "()V", null, null);
-            mv.visitCode();
-            mv.visitFieldInsn(GETSTATIC, "java/lang/System", "out", "Ljava/io/PrintStream;");
-            mv.visitLdcInsn("hello from m");
-            mv.visitMethodInsn(INVOKEVIRTUAL, "java/io/PrintStream", "println", "(Ljava/lang/String;)V", false/*intf*/);
-            mv.visitInsn(RETURN);
-            mv.visitMaxs(3, 1);
-            mv.visitEnd();
-        }
-        {
-            MethodVisitor mv = cw.visitMethod(ACC_PUBLIC | ACC_STATIC, "staticM", "()V", null, null);
-            mv.visitCode();
-            mv.visitFieldInsn(GETSTATIC, "java/lang/System", "out", "Ljava/io/PrintStream;");
-            mv.visitLdcInsn("hello from staticM");
-            mv.visitMethodInsn(INVOKEVIRTUAL, "java/io/PrintStream", "println", "(Ljava/lang/String;)V", false/*intf*/);
-            mv.visitInsn(RETURN);
-            mv.visitMaxs(3, 1);
-            mv.visitEnd();
-        }
-        cw.visitEnd();
-        return cw.toByteArray();
+        byte[] bytes;
+
+        bytes = ClassFile.of().build(ClassDesc.of("IBad"),
```

Comment:
Also, just merger with line 181 or return this result.

Reply:
Done